server/sched.go

```diff
@@ -421,20 +421,17 @@ func (runner *runnerRef) needsReload(ctx context.Context, req *LlmRequest) bool
 	slog.Debug("evaluating already loaded", "model", req.model.ModelPath)
 	runner.refMu.Lock()
 	defer runner.refMu.Unlock()
-	// Ignore the NumGPU settings for comparison
-	optsExisting := runner.Options.Runner
-	optsExisting.NumGPU = -1
-	optsNew := req.opts.Runner
-	optsNew.NumGPU = -1
+
```

Comment:
I think we want to avoid reloading the model if the user specifies -1 for the NumGPU setting on input.

```suggestion
optsExisting := runner.Options.Runner
optsNew := req.opts.Runner
if optsNew.NumGPU < 0 {
  optsExisting.NumGPU = -1
  optsNew.NumGPU = -1
}
```

and leave the check below as it was

Reply:
Do we use optsNew and optsExisting after this change?

Previously, both variables are used in line 437